solr/solrj/src/test/org/apache/solr/client/solrj/impl/LBHttp2SolrClientIntegrationTest.java

```diff
@@ -262,6 +214,31 @@ private LBSolrClient.Endpoint[] bootstrapBaseSolrEndpoints(int max) {
     return solrUrls;
   }
 
+  private void startJettyAndWaitForAliveCheckQuery(SolrInstance solrInstance) throws Exception {
+    final int pollIntervalMs = 50;
+    final int timeoutMs = 10000;
+    int waitSoFarMs = 0;
+
+    try (LogListener logListener = LogListener.info().substring("distrib=false")) {
+      solrInstance.startJetty();
+      while (true) {
+        if (logListener.pollMessage() != null) {
```

Comment:
If we tweak LogListener to use the BlockingQueue semantics of the underlying BlockingQueue, you could poll with a timeout, and thus wait *just* long enough and not longer.  In other words, expose a new pollMessage(time,timeUnit) method and then no need for this loop & interrupt & sleep.  I could imagine more tests using this technique.